src/clojure/nrepl/middleware/auth.clj

```diff
@@ -0,0 +1,40 @@
+(ns nrepl.middleware.auth
+  (:require
+   [nrepl.misc :refer [response-for]]
+   [nrepl.transport :as t])
+  (:import
+   (java.io File FileNotFoundException)))
+
+(defn find-token
```

Comment:
Btw, I think we should cache the result of this function unless we won't the token to be something you can change while the server is running, which doesn't seem very wise to me.

Reply:
Btw, after working on `nrepl.config` I realized that it's bad for middleware/internal functions to lookup config themselves. Finding the token should probably be the responsibility of `lein`/`boot`/`clj` and they should pass this in to nREPL. That adds more flexibility to clients to handle configuration they way they see fit. E.g. I opted not to do env vars in `nrepl.config` as I couldn't figure whether they should have higher or lower precedence than stuff in the global/local config file.

Reply:
Specifically regarding "I think we should cache the result of this function", the current arrangement was intended to capture the token at the time the handler's created, and so you might capture a different token if/when you create a new handler, but only then.

Perhaps you're saying you'd prefer the token to be captured only once for the entire JVM process?  (Oh, and given your other comments, and our unfinished discussion there, I'm going to move this default auth middleware to its own commit, so you can decide independently whether you want to keep it at all.)